### PR TITLE
Add confidence score in healthcare relation and example.

### LINKED ISSUE #19228: [Language - Healthcare Task] REST API Review
Service name: Language Text Analytics - Add confidence scores in healthcare for relations
Key contact for the review:

Engineer: Aurgho Bhattacharjee [aurghob@microsoft.com](mailto:aurghob@microsoft.com)
PM: Peyton Fraser [peytonfraser@microsoft.com](mailto:peytonfraser@microsoft.com)
Whether this is a new or existing API: existing API

Whether or not your service already been deployed: this service has already been deployed, we're just adding in a new feature (field in the response)

A brief description of the material to be reviewed/changed: In the healthcare API, we are adding in a field to share the confidence scores of an extracted relation. (We are already sharing confidence scores for entities.)

### PR DESCRIPTION
**Pull request by aurghob**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by heaths — APPROVED** [review]

**Comment by heaths** [comment]
@lmazuel @weshaggard can one of you merge these despite the duplicate symbol issues with SDK generation. Can we get that fixed? Building all swaggers in a directory is not always practical. We don't ship this way - we shouldn't be validating this way. Do we just need to disable these checks for now until fixed?

/cc @akning-ms

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM, if `confidenceScore` is only in response.

**Comment by weidongxu-microsoft** [comment]
> @lmazuel @weshaggard can one of you merge these despite the duplicate symbol issues with SDK generation. Can we get that fixed? Building all swaggers in a directory is not always practical. We don't ship this way - we shouldn't be validating this way. Do we just need to disable these checks for now until fixed?
> 
> /cc @akning-ms

These sdk automation is not required CI. You should be able to merge the PR even if these fails.

If you want to disable sdk automation, just remove the `swagger-to-sdk` section in
https://github.com/Azure/azure-rest-api-specs/blob/main/dev/cognitiveservices/data-plane/Language/readme.md#swagger-to-sdk

PS, the .NET track1 is known to not working most of the time; and the python is also track1. You can safely remove them.

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzetext.json:899]: Actually, can you add `"x-ms-client-name": "confidence"` as we did in other swaggers? The SDK archboard didn't like "Score" on the properties.

**Review by heaths — CHANGES_REQUESTED** [review]

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (aurghob) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json:899]: I was keeping it consistent with the confidenceScore field in entities. Is the suggestion also to change that field in this case?
Also - should x-ms-client-name only be used for the confidence score fields?

Comment (assafi) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json:899]: @aurghob `x-ms-client-name` just tells the SDK tools to name this field differently when generating the SDKs. The name returned from the service should still be `confidenceScore` for consistency.

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzetext.json:899]: Per offline discussion, seems `confidenceScore` is already used in many places so I'd remain consistent. /cc @KrzysztofCwalina @tg-msft @jsquire about the inconsistency here with recommendations for other Cognitive Services - Language pillar SDKs. Would you all agree it's better to be consistent for the TA SDK than other Language SDKs?

Comment (mikaelsitruk) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json:899]: not sure i get that, language SDK is replacing TA SDK long run. ConfidenceScore should be the preferred term anyway, confidence does not convey a number (e..g score)

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (mikaelsitruk) [inline dev/cognitiveservices/data-plane/Language/analyzetext.json:901]: are we aligned on the type here? are we using double for all confidence score?

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (mikaelsitruk) [inline dev/cognitiveservices/data-plane/Language/analyzetext.json:902]: We should also say the format e..g know many decimal place (should be the same as sentiment)

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (mikaelsitruk) [inline dev/cognitiveservices/data-plane/Language/analyzetext.json:899]: is it regular to see x-ms-client-name? it seems very counterintuitive to for the client on a specific name.

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (mikaelsitruk) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json:901]: resolving since we use double for all confidence score

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (mikaelsitruk) [inline dev/cognitiveservices/data-plane/Language/analyzetext.json:863]: should we make it optional? if a relation does not have a confidence score we shoud not have it in the output, right?

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (heaths) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json:899]: There's no actual "Language SDK". We still have separate TA, CLU, QA, etc. SDKs all as part of the Language unification, but - for various reasons (like size, focus, etc.) we still ship separate SDKs.

So, because TA was and still is a separate SDK, it makes more sense to continue using `ConfidenceScore` as a property instead of just `Confidence` that other, separate SDKs are using.

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (heaths) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json:863]: Correct. If it won't always be in the response, it should not be required.

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (heaths) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json:899]: We need to remove this specific instance but, yes, it's not uncommon to see `x-ms-client-name` or an `x-ms-enum.name` that is different from the enum model name. This is based on SDK architects' feedback. Ideally, REST and the SDK APIs agree, but when it's too late or undesirable to change the REST API, this is the second best option so that all language generators benefit.

Basically, the order of preference for renames should be:

1. Swagger models or properties
2. x-ms-client-name or x-ms-enum.name in swagger
3. Swagger transforms in the specs repo's readme.md
4. Swagger transforms in the individual language repos' readme.md. These are often sharable, so one person could write them and everyone else can copy/paste. 3 is better, but there may be times - like in a crunch - this is easier/faster.
5. For .NET, we have a bunch of attributes we can use to rename classes or properties but I feel we should use this sparingly since it benefits no other languages and, most often, we'll want to be consistent (barring casing) across languages.

Comment (aurghob) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json]: Removed x-ms-client-name as field

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (aurghob) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json]: Removed confidenceScore from required

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (aurghob) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json]: Keeping confidenceScore as the field

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzetext.json:900]: Is this actually true? At least in CLU, it most often more than 2 decimal places. It's probably not even worth mentioning. They get a double back and probably don't need to know more. They should parse it as a double so precision matters little.

**Review by heaths — APPROVED** [review]

**Comment by aurghob** [comment]
Hi @heaths , I don't think I've introduced any new issues here for the dotnet build to fail - I haven't included any new StringIndexType definitions.

File: dev/cognitiveservices/data-plane/Language/analyzetext.json
Comment (aurghob) [reply dev/cognitiveservices/data-plane/Language/analyzetext.json]: In our case, we round the decimal to 2 places and send it. However, I've reverted the description as well.

**Comment by aurghob** [comment]
Addressed and resolved comment @heaths .

**Comment by heaths** [comment]
@aurghob were you going to push those final description changes? I don't see them in this PR. Once done and the tests pass (except those likely to fail for reasons discussed already), I can merge.

**Pull request merged**
